projects/ADuCM3029_demo_cn0531/src/cn0531.c

```diff
@@ -0,0 +1,424 @@
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+* LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#include <stdlib.h>
+#include <math.h>
+#include <stdio.h>
+#include "cn0531.h"
+#include "error.h"
+#include "timer.h"
+
+/**
+ * @brief Help command helper function. Display help function prompt.
+ * @param [in] dev	 	 	 - The device structure.
+ * @param [in] short_command - True to display the long command prompt,
+ *                             false to display the short command prompt.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+static int32_t cn0531_help_prompt(struct cn0531_dev *dev, bool short_command)
+{
+	if (!short_command) {
+		return usr_uart_write_string(dev->cli_desc->uart_device,
+					     (uint8_t*)"\tCN0531 application.\n\r"
+					     "For commands with options as arguments typing the command and 'space' without arguments\n\r"
+					     "will show the list of options.\n\r"
+					     "Available verbose commands.\n\r\n\r");
+	} else {
+		return usr_uart_write_string(dev->cli_desc->uart_device,
+					     (uint8_t*)"\nAvailable short commands:\n\r\n\r");
+	}
+}
+
+/**
+ * @brief Application generic commends. cn0531_help() helper function.
+ * @param [in] dev	 	 	 - The device structure.
+ * @param [in] short_command - True to display the long command prompt,
+ *                             false to display the short command prompt.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+static int32_t cn0531_help_general(struct cn0531_dev *dev, bool short_command)
+{
+	if (!short_command) {
+		return usr_uart_write_string(dev->cli_desc->uart_device,
+					     (uint8_t*)" help                       - Print this tooltip.\n\r"
+					     "                              Example: help\n\r");
+	} else {
+		return usr_uart_write_string(dev->cli_desc->uart_device,
+					     (uint8_t*)" h                - Print this tooltip.\n\r"
+					     "                    Example: h\n\r");
+	}
+}
+
+/**
+ * @brief DAC specific commands tooltip. cn0531_help() helper function.
+ * @param [in] dev	 	 	 - The device structure.
+ * @param [in] short_command - True to display the long command prompt,
+ *                             false to display the short command prompt.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+static int32_t cn0531_help_dac(struct cn0531_dev *dev, bool short_command)
+{
+	if (!short_command) {
+		return usr_uart_write_string(dev->cli_desc->uart_device,
+					     (uint8_t*)" dac_reg_read <addr>        - Read a DAC register.\n\r"

```

Comment:
This strings can be added as static global variables or defines. You will have more space for each line and the logic of the code will be easy to read.